### ISSUE TITLE
How to detect macos architecture inside setup.py

### ISSUE THREAD

**Issue by reynoldsnlp** [pypa/cibuildwheel#1537]
### Description

I would like to change paths for `include` and `libs` in `setup.py`, depending on whether `cibuildwheel` is building for `x86_64` or `arm64` on MacOS. Is there an environment variable or something similar that I can depend on?

I've tried using `platform.version()` and `platform.uname()`, but they always show `x86_64`.

**Comment by reynoldsnlp** [comment]
Looks like it can be accessed by inspecting `sys.executable`. Is there a better way to do this?

**Comment by joerick** [comment]
`platform.machine()` is the standard way that cibuildwheel does it.

**Comment by joerick** [comment]
Oh, you want this to support cross-builds. Simplest thing would be to check the _PYTHON_HOST_PLATFORM env var, and fall back to `platform.machine()` if it's not set. See cibuildwheel/macos.py for values. 

There might be a cleaner method using the `sysconfig` built in module.

**Comment by henryiii** [comment]
Keep in mind, that tells you about the Python interpreter you are compiling with if `_PYTHON_HOST_PLATFORM` isn't set. Specifically, `is_universal2` could be true on either x86 or ARM if you have a universal build of Python. That's fine, just remember to handle that correctly in your setup.py. :)

**Comment by joerick** [comment]
Yep. Though I think that detail echos what most build backends would do, too?

**Comment by reynoldsnlp** [comment]
This works well for me. Is it something that's worth adding to the docs? If so, I would be happy to submit a PR. Otherwise, feel free to close the issue.

**Comment by joerick** [comment]
We could add a note to the FAQ around this area - https://cibuildwheel.readthedocs.io/en/stable/faq/#how-to-cross-compile "Detecting the target architecture when cibuildwheel is cross-compiling" - noting the _PYTHON_HOST_PLATFORM mechanism and the `sysconfig.get_platform()` method. If you have time that would be much appreciated @reynoldsnlp !

**Comment by JamesTheAwesomeDude** [comment]
I'm also curious about this in non-Mac cases.

Say I'm cross-compiling for aarch64 on Linux \(with an x86_64 host\).

I want to leave *out* a bunch of `ext_modules` that compile AVX2 libraries, since there's obviously 0% chance they would be appropriate for the target, even if there's a 99% chance the *host* will be able to run it.

Is a call to `sysconfig.get_platform` going to return the target architecture, or the host architecture?

**Comment by henryiii** [comment]
We don’t support cross compiling on Linux, only windows and macOS. Linux has other issues (like being forced to use the “system” compiler which is too old to support the new architecture), so it’s generally not useful to pursue traditionally. The Zig compiler sounds like a much more likely path to try.

I think _PYTHON_HOST_PLATFORM works on any system, though.

**Comment by joerick** [comment]
+1 to all Henry said. 

Just a little note on terminology-

> Is a call to `sysconfig.get_platform` going to return the target architecture, or the host architecture?

It'll return the 'target' i.e. the platform the wheel you're building is gonna run on. Confusingly, in compiler terminology, that's also referred to as the 'host' platform (the machine that does the compiling is called the 'build' system). I suspect that's why the environment variable is called _PYTHON_HOST_PLATFORM.

**Comment by henryiii** [comment]
There's also CIBUILDWHEEL_BUILD_IDENTIFIER now (well, in 4.0.0rc2) too.
